packages/@react-stately/color/src/useColorFieldState.ts

```diff
@@ -113,22 +114,24 @@ export function useColorFieldState(
     }
 
     // if it failed to parse, then reset input to formatted version of current number
-    if (parsed.current == null) {
+    if (parsedColor == null) {
       setInputValue(colorValue ? colorValue.toString('hex') : '');
       return;
     }
 
-    safelySetColorValue(parsed.current);
+    let newColor = safelySetColorValue(parsedColor);
     // in a controlled state, the numberValue won't change, so we won't go back to our old input without help
-    let newColorValue = '';
-    if (colorValue) {
-      newColorValue = colorValue.toString('hex');
+    if (value !== undefined || colorsAreEqual(newColor, colorValue)) {
+      setForceSync(true);
     }
-    setInputValue(newColorValue);
-  };
-
-  let increment = () => {
-    let newValue = addColorValue(parsed.current, step);
+  }, [inputValue, safelySetColorValue, setInputValue, value, colorValue, parsedColor]);
+
+  let spinRef = useRef(parsedColor);
+  useEffect(() => {
```

Comment:
translation to useEffect should be fine, the spinning of the spin button is slower than a raf at 60ms, and unless you're a gamer going for a world record, you're unlikely to double click even close to a 16ms speed

If we wanted to have this ok for < 16ms responses, I'm not sure what we could do for controlled state colorfields because we wouldn't know for a render cycle if they accepted the new value or not